src/Temporalio/Worker/TemporalWorkerOptions.cs

```diff
@@ -46,6 +46,22 @@ public TemporalWorkerOptions(string taskQueue)
         /// Gets or sets the workflow types.
         /// </summary>
         public IList<Type> Workflows { get; set; } = new List<Type>();
+
+        /// <summary>
+        /// Gets or sets additional activity definitions.
+        /// </summary>
+        /// <remarks>
+        /// Unless manual definitions are required, users should use <see cref="Activities" />.
+        /// </remarks>
+        public IList<Activities.ActivityDefinition> AdditionalActivityDefinitions { get; set; } = new List<Activities.ActivityDefinition>();
```

Comment:
nit: It seems confusing to have a distinction between `Activities` and `AdditionalActivityDefinitions`. I understand that you factored the API this way so that you can have a list-based mechanism for registering implicitly named activities (and workflows) and explicitly named activities. However, the benefit of this design is not clear to me and I worry that the average developer wouldn't understand the motivation. I suspect it would be more natural to have two overloaded `AddActivity()` methods (`AddActivity(Delegate)` and `AddActivity(string, Delegate)`) and then a single `IReadOnlyList<ActivityDefinition> Activities { get; }` property to enable enumerating the definitions.

Reply:
I create my worker options like `new() { Activities = { ActivityClass.ActivityMethod } }`. I also intentionally made this unwieldy to discourage its use. Are there other ways to solve my goals of 1) making delegates with attributes the obvious preferred approach, and 2) making delegates with attributes easy to set as properties? I guess I could make a custom collection. Would that be a better approach?

I was expecting, with one of my samples, to show how to set activities as fully qualified method names using https://learn.microsoft.com/en-us/dotnet/core/extensions/configuration so I was hoping to keep everything properties I could one-day serialize into from JSON (granted I know that I'd have to manually resolve method references from strings during deserialization).

Reply:
I see. Supporting a JSON representation of worker options is an interesting idea. I can't immediately think of a better approach given these goals.